### PR TITLE
Redundant parameter eager

### LINKED ISSUE #11216: Redundant query parameter in many-to-many relationships in entity
##### **Overview of the issue**

When we generate an entity which contains many-to-many relationship with another entity, JHipster creates a query parameter for getAll endpoints called `eagerload`, but it's not used anywhere in EntityResource.java. 

There is already a logic to use either `findAllWithEagerRelationships` or `findAll` in the ejs template based on if the entity has a many-to-many relationship or not, so not sure why there is also an unused `eagerload` query parameter in the endpoints. 

If we want to give the option to load it eager or not during the runtime, shouldn't the generated Java code have:
```
if (eagerload) {
    return entityRepository.findAllWithEagerRelationships();
} else {
    return entityRepository.findAll();
}
```
?

##### **Motivation for or Use Case**

To make it configurable to eager or lazy load many-to-many relationships during runtime

##### **Reproduce the error**

1- Generate a microservice
2- Add an entity that has many-to-many relationship with another entity
3- Check the generated EntityResource.java file, getAll method have `eagerload` parameter but it isn't used.

##### **Suggest a Fix**

We should either remove `eagerload` parameter or use it in the getAll method.

##### **JHipster Version(s)**

6.5.1

##### **JHipster configuration**

We used this sample JDL https://github.com/jhipster/jdl-samples/blob/master/simple-online-shop.jh

##### **Browsers and Operating System**

MacOS Catalina

-   [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by seraphinandrieux**
PR to fix #11216 

I chose to remove the parameter eager load inside the generator. I based my solution on the logic already made with entities which have a Many to Many relation.
Inside the generator each time we have a many to many relation between two entities, the ejs chooses to use the method `findAllWithEager()`.
So I removed the query parameter `eagerload` which is useless if we follow the logic used someplace else.

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs [hunk 1]
```diff
@@ -49,23 +49,19 @@ _%>
...
         return <%= entityListToDtoListReference %>(<%= entityInstancePlural %>);<% } else { %>
         return <%= entityInstance %>Repository.<% if (fieldsContainOwnerManyToMany) { %>findAllWithEagerRelationships<% } else { %>findAll<% } %>();<% } %>
     <%_ } if (pagination !== 'no') { _%>
-    public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(Pageable pageable<% if (reactive) { %>, ServerHttpRequest request<% } %><% if (fieldsContainNoOwnerOneToOne) { %>, @RequestParam(required = false) String filter<% } %><% if (fieldsContainOwnerManyToMany) { %>, @RequestParam(required = false, defaultValue = "false") boolean eagerload<% } %>) {<%- include('get_all_stream_template', {viaService: viaService}); -%>
+    public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(Pageable pageable<% if (reactive) { %>, ServerHttpRequest request<% } %><% if (fieldsContainNoOwnerOneToOne) { %>, @RequestParam(required = false) String filter<% } %>) {<%- include('get_all_stream_template', {viaService: viaService}); -%>
```
Comment (MathieuAA) [inline generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs]: I'm not so sure about this. Previously, the default was `false`, right? So removing it but making fetching eager relationships by default kinda makes it "always true".
I feel like this is a breaking change, kinda.

Comment (seraphinandrieux) [reply generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs]: Indeed my bad, I'm updating the code to take account of this default value. 
Thanks for your review !

Comment (MathieuAA) [reply generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs]: Maybe we should make fetching eager relationships the default for the next major version (v7). Again, not so sure about this either :(

Comment (seraphinandrieux) [reply generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs]: Do you recommend me to put back the `eagerload` query argument and keep the logic used ?

**Pull request closed without merging**